### ISSUE TITLE
Stack smashing on 16 bit png images, plus minor warning on interlaced images

### ISSUE THREAD

**Issue by Aicaya** [stoeckmann/xwallpaper#16]
> Distro: Arch Linux

If an image is set to 16-bit, xwallpaper will crash with the following error:

```
*** stack smashing detected ***: terminated
Bad system call (core dumped)
```

Converting it to 8-bit with Gimp (Image > Precision > 8 bit integer) or converting it to jpeg makes it work again, but a bug is a bug so here's a report.

---

On a separate note, interlaced png files produce the following warning message:

```
libpng warning: Interlace handling should be turned on when using png_read_image
```

It still sets the wallpaper without issue so it's potentially harmless, but I figured I'd report it anyway.

(also pardon the edits—I misread a closed issue and thought the first half was a duplicate)

**Comment by stoeckmann** [comment]
What a horrible mistake. Definitely a big thank you for reporting it. The 16 bit code for PNG was working correctly for grayscale, but the "skip the upper 8 bits" flag was not set for regular files.

I have just pushed changes for both issues and will immediately release a new version.

Thanks again! Do you want to request a CVE for this or should I do so?

**Comment by Aicaya** [comment]
Belated but I'm not familiar with CVEs, so best leave it to you if you feel it necessary.

Thanks for the response!

**Comment by stoeckmann** [comment]
Okay, I'll take care. Closing this issue now since it's fixed in 0.6.4. If anything else pops up, please let me know. :)
